### PR TITLE
feat: partial metadata projection

### PR DESCRIPTION
**Pull request by loferris**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/nodejs-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

This PR updates the documentation to specify how to get a partial metadata projection via the `table.get()` method, and clarified the difference between this method and `table.getMetadata()` in terms of being selective about metadata retrieved. A system test for this usage of `table.get()` has been added, as well as an expansion of the `table.getMetadata()` system test to help illustrate the differences between these two methods. This PR reflects an update to the BigQuery API across all languages, not a GH issue for this repo.

### CODE REVIEW

File: system-test/bigquery.ts [hunk 1]
```diff
@@ -888,9 +887,25 @@ describe('BigQuery', () => {
       const description = 'catsandstuff';
       await table.setMetadata({description});
       const [metadata] = await table.getMetadata();
+      const metadataProps = Object.values(metadata);
+      assert.strictEqual(metadataProps.length, 18);
```
Comment (alvarowolfx) [inline system-test/bigquery.ts:891]: I think checking for the number of attributes here can lead the test to be flaky in the long term, as every time that the service team adds a new field, the test is going to break.

File: system-test/bigquery.ts [hunk 2]
```diff
@@ -888,9 +887,25 @@ describe('BigQuery', () => {
...
+    it('should set & get partial metadata', async () => {
+      const options = {
+        view: 'BASIC',
+      };
+      const [basicMetadata] = await table.get(options);
+      const basicMetadataProps = Object.values(
+        Object.keys(basicMetadata.metadata)
+      );
+
+      assert.strictEqual(basicMetadataProps.length, 10);
```
Comment (alvarowolfx) [inline system-test/bigquery.ts:904]: Same comment around testing for the amount of fields

**Review by alvarowolfx — CHANGES_REQUESTED** [review]
Checking the implementation guide, I think a good idea to test the different between getting a basic view vs full view is to check for the `lastModifiedTime` and/or `numBytes`, as in the basic view, those attributes are not computed by the backend.

https://github.com/googleapis/google-cloud-go/pull/6186/files#diff-62ca20d473019007434974e167cb4ad4be57a9f5b8b4ba834dfcf7abe3ee4f76

File: system-test/bigquery.ts (hunk 1 above)
Comment (loferris) [reply system-test/bigquery.ts:891]: Good point!

**Comment by loferris** [comment]
> Checking the implementation guide, I think a good idea to test the different between getting a basic view vs full view is to check for the `lastModifiedTime` and/or `numBytes`, as in the basic view, those attributes are not computed by the backend.
> 
> https://github.com/googleapis/google-cloud-go/pull/6186/files#diff-62ca20d473019007434974e167cb4ad4be57a9f5b8b4ba834dfcf7abe3ee4f76

I'm already checking against `numBytes` in the new system test so I can add `lastModifiedTime` as well.

**Review by alvarowolfx — APPROVED** [review]

**Pull request merged**